### PR TITLE
[FIX] sale_timesheet: fix task creation when no tasks

### PR DESCRIPTION
**Pull request by agr-odoo**
Create a sale order with a service configured with
- Service Invoicing Policy: Timesheets on tasks
- Service Tracking: Create a new project but no task
Save and confirm. Access the project and the tasks via smart button.
Create a new task.

The task will not be saved under the current project because there is no
data reference in context.

opw-2355547

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by agr-odoo** [comment]
@nim-odoo

File: addons/sale_timesheet/models/project_overview.py [hunk 1]
```diff
@@ -424,6 +424,8 @@ def _plan_get_stat_button(self):
 
         if len(task_projects_ids) == 1:
             tasks_context = {**tasks_context, 'default_project_id': task_projects_ids[0]}
+        elif not task_projects_ids and len(self) == 1:
+            tasks_context = {**tasks_context, 'default_project_id': self.id}
```
Comment (yelizariev) [inline addons/sale_timesheet/models/project_overview.py:428]: Maybe you can check ``len(self) == 1`` first, i.e.

      if len(self.ids) == 1:
            tasks_context = {**tasks_context, 'default_project_id': self.id}
      elif len(self.ids):
        # filter out all the projects that have no tasks
        ...

Comment (agr-odoo) [reply addons/sale_timesheet/models/project_overview.py:421]: thanks for the suggestions, indeed some refactoring was possible to avoid redundant checks

Comment (yelizariev) [reply addons/sale_timesheet/models/project_overview.py:421]: Not sure that you can remove previous logic. You should address a question to PM, if you want to change it. Or at least update commit message. 
New version doesn't work in the same way in scenario when you remove filter in "Project overview": if only one project has tasks, then before you commit default project is that project, but after your commit there is no default project

There is a missed step on the picture (between 1 and 2 steps): add a fitler that search results for severals projects and only one of those projects has tasks
screenshot-4541009-13-0 runbot50 odoo com-2020 10 13-17_29_25

**Review by yelizariev — APPROVED** [review]
Should be fine now

**Comment by agr-odoo** [comment]
@nim-odoo

**Review by nim-odoo — APPROVED** [review]
robodoo delegate+

**Comment by agr-odoo** [comment]
@robodoo r+

**Pull request closed without merging**